doc/guides/prog_guide/cnet.rst

```diff
@@ -0,0 +1,400 @@
+
+.. note:: CNET Stack *TCP* support is **experimental** at this time for release 22.08.
+
+.. _figure_cndp_system_overview:
+
+.. figure:: img/cndp_system_overview.*
+
+   CNDP System Overview
+
+A CNDP application is running in user-space in a container or bare-metal environments and provides
+a set of libraries for the application developer to quickly create a *Cloud Native* application for
+more details and information about CNDP please refer to the :ref:`CNDP Overview <CNDP_Overview>`.
+
+A number of libraries are available in CNDP and some are used in the CNET stack to provide better
+performance and functionality for the application developers please refer to
+:ref:`CNDP Components <CNDP_Components>` as needed.
+
+.. note:: The *cnet-graph* example is a reasonable source on how to create an application that
+  uses the CNET graph nodes. The *l3fwd-graph* example uses a different set of graph nodes then
+  CNET stack, but provides another solution for a very simple packet processing application.
+
+CNET POD/Container Overview
+---------------------------
+
+The following CNET stack overview shows the POD/Container view and where CNDP components will
+be placed in a POD/Container. The *Sidecar* container is used to configure and obtain statistics
+for the CNDP application. The *Sidecar* container is *optional* and is not used in all applications
+use cases.
+
+.. _figure_cnet_overview:
+
+.. figure:: img/cnet_overview.*
+
+   CNET POD/Container Overview
+
+The CNET stack splits some of the processing of packets between the Linux kernel and the CNDP
+application. Packets like ARP requests are handled by the Linux stack and if an ARP request is
+received by the application it will need to be punted to the kernel for more processing. Other
+packets not being processed by the CNET stack will be dropped or punted to the Linux kernel
+via the *punt_kernel* node.
+
+Because packets like ARP requests are being processed by the Linux stack and updating the Linux
+ARP table we need a method to get this information to the CNET stack. The method being used
+is the *Netlink* messages. When the kernel learns a new ARP address a *Netlink* message is send
+and the CNET stack receives these messages and updates the CNET internal tables.
+
+A number of different *Netlink* messages are processed by the CNET stack to updatig ARP, routes and
+interface information. This to allows CNET to learn about the system and allow the operator
+to use standard Linux command line and tools i.e. ifconfig, ip, route, ... to configure the CNET stack.
+The *Netlink* message processing is handled by a different thread within the application is not
+directly attached to the CNET graph node instances.
+
+CNET Graph nodes
+----------------
+
+The CNET library contains support for UDP/TCP/IPv4 and IPv6 in the future. We have a number
+of different graph nodes in CNET and creating your own graph node is very straight forward
```

Comment:
end the sentence after 'straight forward'. Start a new sentence with "Please refer..."